Source/CardView.m

```diff
@@ -187,21 +163,32 @@ - (void)tapAction:(UITapGestureRecognizer *)tapRecognizer {
     }
 }
 
-- (void)swipeRightAction:(UISwipeGestureRecognizer *)swipeRightRecognizer {
-    if ([self.delegate respondsToSelector:@selector(cardRemoveRequested:)]) {
-        [self.delegate cardRemoveRequested:self];
-    }
-}
-
-- (void)swipeUpAction:(UISwipeGestureRecognizer *)swipeRightRecognizer {
-    if ([self.delegate respondsToSelector:@selector(cardTitleSwipeUp:)]) {
-        [self.delegate cardTitleSwipeUp:self];
-    }
-}
-
-- (void)swipeDownAction:(UISwipeGestureRecognizer *)swipeRightRecognizer {
-    if ([self.delegate respondsToSelector:@selector(cardTitleSwipeDown:)]) {
-        [self.delegate cardTitleSwipeDown:self];
+- (void)panAction:(UIPanGestureRecognizer *)panRecognizer {
+    static CGPoint originalPoint;
+
+    switch (panRecognizer.state) {
+        case UIGestureRecognizerStateBegan:
+            originalPoint = [panRecognizer locationInView:self.superview];
+            if ([self.delegate respondsToSelector:@selector(cardTitlePanDidStart:)]) {
+                [self.delegate cardTitlePanDidStart:self];
+            }
+            break;
+        case UIGestureRecognizerStateChanged: {
+            CGPoint point = [panRecognizer locationInView:self.superview];
+            CGPoint delta = CGPointMake(point.x - originalPoint.x, point.y - originalPoint.y);
+            if ([self.delegate respondsToSelector:@selector(card:titlePannedByDelta:)]) {
+                [self.delegate card:self titlePannedByDelta:delta];
+            }
+            break;
+        }
+        case UIGestureRecognizerStateCancelled:
+        case UIGestureRecognizerStateEnded:
+            if ([self.delegate respondsToSelector:@selector(cardTitlePanDidFinish:withVelocity:)]) {
+                [self.delegate cardTitlePanDidFinish:self withVelocity:[panRecognizer velocityInView:self.superview]];
+            }
+            break;
+        default:
```

Comment:
Please remove the default, better to add an empty implementation for the missing cases. Xcode will throw a warning if new cases are added/removed/renamed. This is a general "best practice".